docs/internal/investigations/remote-runtime-configuration-dapr/poc/dapr-poc-scenario.md

```diff
@@ -0,0 +1,86 @@
+
+1. Control Plane admin sets the configuration for two Runtimes.
+2. Runtime Agents from both Runtimes fetch the configuration and create the Dapr Component resources.
+3. Control Plane admin changes the configuration.
+4. Agents fetch the new configuration, apply it, and restart the Pods.
+5. Both Runtimes use the new configuration.
+
+As it's the Proof of Concept stage, we can use some temporary solutions, such as:
+- Configuration held in memory instead of a database
+- Pods being restarted by the Agent instead of a more complex solution
+
+### Initial architecture
+
+![Initial Architecture Diagram](./stage-1.svg)
+
+The image depcts the initial configuration of the components which will be deployed.
+
+Tools:
+- docker [https://www.docker.com/](https://www.docker.com/)
+- k3d v3 [https://github.com/rancher/k3d](https://github.com/rancher/k3d)
+- kubectl [https://kubernetes.io/docs/tasks/tools/install-kubectl/](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- helm 3 [https://helm.sh/](https://helm.sh/)
+
+### Control Plane deployment
+
+In the first step we deploy the Control Plane cluster which hosts the Runtime Director component as well as two Redis instances for the purpose of persising the state by the Runtimes.
+
+To deploy the Control Plane cluster run the follwing command
+```
+./setup-cp.md
+```
+
+You can watch the installation progress by looking at the statuses of the pods being deployed in the cluster
+
+```
+watch kubectl --context=k3d-cp get po --all-namespaces
+```
+
+### Runtimes deployment
+
+Next step is to deploy two runtimes that will consume the Control Plane API and persis the state in the Redis instances configured by the central Control Plane.
+
+To deploy the first Runtime cluster run the following command. It will create the cluster and deploy a basic set of components including the nodeapp that demoes the use-case with the state store.
+```
+./setup-runtime1.sh
+```
+
+You can watch the installation progress by looking at the statuses of the pods being deployed in the cluster. On that stage there should be nodeapp pod deployed but the Dapr sidecar should not be injected yet.
+
+```
+watch kubectl --context=k3d-runtime1 get po --all-namespaces
+```
+
+Next, deploy the agent component using the following command. It will deploy the agent that connects to the Control Plane cluster and fetches the configuration for the given runtime.
+
+```
+./setup-runtime1-agent.sh
+```
+
+You can watch the installation progress by looking at the statuses of the pods being deployed in the cluster. After successful deployment of the agent component it should fetch the confgiuration and apply it on the cluster. The nodeapp pod should be recreated with the Dapr sidecar injected.
+
+```
+watch kubectl --context=k3d-runtime1 get po --all-namespaces
+```
+
+In order to run second runtime cluster please run all aboved steps using `runtime2` scripts.
```

Comment:
could we add a short chapter with resetup-runtime1.sh that will switch to redis2? I understand we dont need to resetup agent1 right?